src/hooks/api/authenticated-query.js

```diff
@@ -129,11 +131,43 @@ const useAuthenticatedMutation = ({
   return useMutation(innerMutationFn, configuration);
 };
 
-const useAuthenticatedQuery = (options) => {
+const useAuthenticatedQuery = ({ mockData, ...options } = {}) => {
   if (!!options.req && !!options.queryClient && typeof window === 'undefined') {
     return prefetchAuthenticatedQuery(options);
   }
 
+  const [randomMockData, setRandomMockData] = useState();
+  const mockDataArray = castArray(mockData);
+
+  const getNewMockData = () => {
+    if (mockDataArray.length === 1) return mockDataArray[0];
+    return mockDataArray[Math.floor(Math.random() * mockDataArray.length)];
+  };
+
+  const refetch = () => setRandomMockData(getNewMockData());
+
+  useEffect(() => {
+    if (
+      mockData &&
```

Comment:
```suggestion
      !!mockData &&
```

Just to be sure I guess